### ISSUE TITLE
Copying a spy.StructDict returns a dict

### ISSUE THREAD

**Issue by dfsp-spirit** [esi-neuroscience/syncopy#394]
**Describe the bug**
Copying a spy.StructDict returns a dict

**To Reproduce**
```python
import syncopy as spy
dd = spy.StructDict()
type(dd)
Out[18]: syncopy.shared.tools.StructDict
type(dd.copy())
Out[19]: dict
```

**Expected behavior**
```python
type(dd.copy())
Out[19]: syncopy.shared.tools.StructDict
```

**System Profile:**
 - OS: Ubuntu 22.04
 - Please paste the output of the following command here 

```python
   import syncopy as spy; spy.AnalogData().log
>>> SyNCopy v. v2022.08-623-g78070d6 <<< 
Created: Wed Dec 14 12:35:23 2022 
System Profile: 
3.8.13 (default, Mar 28 2022, 11:38:47) 
[GCC 7.5.0] 
ACME:  2022.7
Dask:  2022.05.0
NumPy: 1.21.5
SciPy: 1.7.3
--- LOG ---
|=== user@box: Wed Dec 14 12:43:30 2022 ===|
	__init__: created AnalogData object
  ```

**Additional Information**

- Tested on current dev branch [7f31377](https://github.com/esi-neuroscience/syncopy/commit/7f31377373340d4c652de04d6bf9a6210ec0edbe)
- Workaround: wrap the copy in spy.StructDict constructor that takes a dict:

```
type(spy.StructDict(dd.copy()))
Out[22]: syncopy.shared.tools.StructDict
```

**Comment by tensionhead** [comment]
This got addressed with #400 right?!

**Comment by dfsp-spirit** [comment]
Yes, closing as done.
